src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -5610,8 +5610,8 @@ GetPoolinfoViaCatalog(int64 nodeId)
 	char *poolinfo = NULL;
 
 	/* set scan arguments */
-	ScanKeyInit(&scanKey[0], nodeIdIdx, BTEqualStrategyNumber, F_INT4EQ,
-				Int32GetDatum(nodeId));
+	ScanKeyInit(&scanKey[0], nodeIdIdx, BTEqualStrategyNumber, F_INT8EQ,
```

Comment:
This is reading from pg_dist_poolinfo.nodeid, which is marked as int.

It seems the wrong thing is `GetPoolinfoViaCatalog` getting int64. It should get int32